python/src/aiconfig/editor/client/src/components/prompt/prompt_input/PromptInputRenderer.tsx

```diff
@@ -3,49 +3,95 @@ import { memo, useState } from "react";
+import { ErrorBoundary, useErrorBoundary } from "react-error-boundary";
+import { Text } from "@mantine/core";
+import JSONRenderer from "../../JSONRenderer";
+import JSONEditorToggleButton from "../../JSONEditorToggleButton";
 
 type Props = {
   input: PromptInput;
   schema?: PromptInputSchema;
   onChangeInput: (value: PromptInput) => void;
 };
 
+type ErrorFallbackProps = {
+  input: PromptInput;
+  toggleJSONEditor: () => void;
+};
+
+function InputErrorFallback({ input, toggleJSONEditor }: ErrorFallbackProps) {
+  const { resetBoundary: clearRenderError } = useErrorBoundary();
+  return (
+    <Flex direction="column">
+      <Text color="red" size="sm">
+        Invalid input format for model. Toggle JSON editor to update
+      </Text>
+      <JSONRenderer content={input} />
+      <Flex justify="flex-end">
+        <JSONEditorToggleButton
+          isRawJSON={false}
+          setIsRawJSON={() => {
+            clearRenderError();
+            toggleJSONEditor();
+          }}
+        />
+      </Flex>
+    </Flex>
+  );
+}
+
 export default memo(function PromptInputRenderer({
   input,
   schema,
   onChangeInput,
 }: Props) {
   const [isRawJSON, setIsRawJSON] = useState(false);
+  const rawJSONToggleButton = (
+    <Flex justify="flex-end">
+      <JSONEditorToggleButton
+        isRawJSON={isRawJSON}
+        setIsRawJSON={setIsRawJSON}
+      />
+    </Flex>
+  );
   return (
     <>
       {isRawJSON ? (
-        <PromptInputJSONRenderer input={input} onChangeInput={onChangeInput} />
-      ) : schema ? (
-        <PromptInputSchemaRenderer
-          input={input}
-          schema={schema}
-          onChangeInput={onChangeInput}
-        />
+        <>
+          <PromptInputJSONRenderer
+            input={input}
+            onChangeInput={onChangeInput}
+          />
+          {rawJSONToggleButton}
+        </>
       ) : (
-        <PromptInputConfigRenderer
-          input={input}
-          onChangeInput={onChangeInput}
-        />
-      )}
-      <Flex justify="flex-end">
-        <Tooltip label="Toggle JSON editor" withArrow>
-          <ActionIcon onClick={() => setIsRawJSON((curr) => !curr)}>
-            {isRawJSON ? (
-              <IconBracesOff size="1rem" />
+        <ErrorBoundary
+          fallbackRender={() => (
+            <InputErrorFallback
+              input={input}
+              toggleJSONEditor={() => setIsRawJSON(true)}
+            />
+          )}
+        >
+          <>
+            {schema ? (
+              <PromptInputSchemaRenderer
+                input={input}
+                schema={schema}
+                onChangeInput={onChangeInput}
+              />
             ) : (
-              <IconBraces size="1rem" />
+              <PromptInputConfigRenderer
+                input={input}
+                onChangeInput={onChangeInput}
+              />
             )}
-          </ActionIcon>
-        </Tooltip>
-      </Flex>
+            {rawJSONToggleButton}
+          </>
```

Comment:
Nit: lots of nested ternary expressions. Could we move this into it's own helper component like `<PromptInputRendererForNonJson>` or something?